### DISCUSSION TITLE
Next.js 13 new router - shallow routes support?

### DISCUSSION THREAD

**Discussion by vucinatim** [vercel/next.js#42242] [App Router]
What is the recommended way to handle shallow routing within the app directory? (If there even is a way)
Is the new router from next/navigation getting some upgrades that allow similar state managing with urls like the old router has?

**Comment by premier213** [comment]
yes, shallow migrate with prefetch function.
actual this performance is better than shallow routing

**Reply by vucinatim** [reply]
But what if i wish to store state in the url parameters.. So that individual components can read from those params and change them one by one without refreshing the page. Also why does the new router.push() only accept an href string instead of an URL object aswell?

**Reply by premier213** [reply]
I think changing the server component & client component using shallow is not recommended, for this added `{ params, searchParams }` to default props in the page function

**Reply by vucinatim** [reply]
I see. Well thank you for replying.
I have a specific case where i have a DateRangePicker component where the user can choose a date.
But if the user decides to change the check_in and check_out parameters in the url i want this component to instantly display those changed values. This makes these kind of urls also useful for sharing the specific page configuration with others.
If you have any recommendations on how to aproach this better i would be happy to know.

**Reply by premier213** [reply]
ok, I think this hook is suitable for this work `useSearchParams()` and set method.
